File: JS/FizzBuzz.js
const FizzBuzz = () => {
    fizzBuzzArr = [];
    for(let i = 0; i < 101; i++) {
        if(i % 15 === 0){
            fizzBuzzArr.push("FizzBuzz")
        } else if(i % 3 === 0) {
            fizzBuzzArr.push("Fizz")
        } else if (i % 5 === 0) {
            fizzBuzzArr.push("Buzz")
        } else {
            fizzBuzzArr.push(i)
        }
    }
    return fizzBuzzArr;
};

FizzBuzz();

File: Ruby/fizzbuzz.rb
# Simple solution with DRY principles in mind
1.upto(15) do |i|
  fizz = (i % 3 == 0)
  buzz = (i % 5 == 0)

  puts case
    when fizz && buzz then 'FizzBuzz'
    when fizz then 'Fizz'
    when buzz then 'Buzz'
    else i
  end
end

# Solution with parameterization
def fizzbuzz(range, triggers)
  range.map do |i|
    result = ''
    triggers.each do |(text, predicate)|
      result << text if predicate.call(i)
    end
    result == '' ? i : result
  end
end

puts fizzbuzz(1..15, [
  ['fizz', ->(i){i%3==0}],
  ['buzz', ->(i){i%5==0}],
])
